### ISSUE TITLE
Make the build version qualifier aware

### ISSUE THREAD

**Issue by graphaelli** [elastic/apm-server#1348]
Allow build time configuration of version qualifier so that build candidates can be promoted to ga without being rebuilt.

Setting `BEAT_VERSION` gets pretty close, eg:

```
$ BEAT_VERSION=7.0.1 mage package
$ find build/distributions/ -type f | head
build/distributions//apm-server-7.0.1-amd64.deb
build/distributions//apm-server-oss-7.0.1-i686.rpm.sha512
build/distributions//apm-server-oss-7.0.1-windows-x86_64.zip.sha512
build/distributions//apm-server-7.0.1-linux-x86_64.tar.gz.sha512
build/distributions//apm-server-7.0.1-i386.deb
...
```

But the version embedded in the binary still reports `7.0.0-alpha1`.  Need to check the bundled readme / docs too, still.

**Comment by graphaelli** [comment]
Spent some time [on this today](https://github.com/elastic/apm-server/compare/master...graphaelli:build-qualifier) with an approach that looks more like how snapshots are handled.

In practice, this works like (`mage -clean`s omitted):
```
# default on 6.4
$ mage build && ./apm-server version
apm-server version 6.4.1 (amd64), libbeat 6.4.1 [a7b71d7a521d44a7fe400207753b457e393acf37 built 2018-09-07 20:51:04 +0000 UTC]

# default on master
$ mage build && ./apm-server version
apm-server version 7.0.0-alpha1 (amd64), libbeat 7.0.0-alpha1 [fb0082ae28b14a74248a466ad9417dc7289ec5c6 built 2018-09-07 20:52:34 +0000 UTC]

# alpha2 on master
$ BEAT_VERSION_QUALIFIER=alpha2 mage build && ./apm-server version
apm-server version 7.0.0-alpha2 (amd64), libbeat 7.0.0-alpha2 [fb0082ae28b14a74248a466ad9417dc7289ec5c6 built 2018-09-07 20:52:53 +0000 UTC]

# release on master
$ BEAT_VERSION_QUALIFIER= mage build && ./apm-server version
apm-server version 7.0.0 (amd64), libbeat 7.0.0 [fb0082ae28b14a74248a466ad9417dc7289ec5c6 built 2018-09-07 20:53:01 +0000 UTC]
```

**Comment by graphaelli** [comment]
closed by #1399
